### ISSUE TITLE
Connection issue after azure-kusto-data==4.1.0

### ISSUE THREAD

**Issue by Gjermund06** [Azure/azure-kusto-python#468]
I have been using azure-kusto-data==4.0.2 until recently. Created a new virtual environment and got latest version (4.1.3) installed and after that no quieries to adx works anymore. 

Tested on Python 3.8 - 3.11.
For azure-kusto-data < 4.1.0 quiery return data, but for >=4.1.0 query fails.

I have adopted example in https://github.com/Azure/azure-kusto-python/blob/master/azure-kusto-data/tests/sample.py with credentials.

Below is the example code i'm running and the results from version below and above vesion 4.1.0. For V4.0.2, I got data as expected but for v4.1.0, firstly i got a 400 response and an errormessage that the server response contains no data.

I have tried with both az-login auth and AAD application auth, the results is the same. 

```
import logging

logging.basicConfig(level=logging.DEBUG)
mylog = logging.getLogger('main')


from datetime import timedelta

from azure.kusto.data import KustoClient, KustoConnectionStringBuilder, ClientRequestProperties
from azure.kusto.data.exceptions import KustoServiceError
from azure.kusto.data.helpers import dataframe_from_result_table

######################################################
##                        AUTH                      ##
######################################################

# Note that the 'help' cluster only allows interactive
# access by AAD users (and *not* AAD applications)
cluster = "https://adxvipanalytics.westeurope.kusto.windows.net"

# In case you want to authenticate with AAD application.
client_id = "28a***"
client_secret = "EZ**"

# read more at https://docs.microsoft.com/en-us/onedrive/find-your-office-365-tenant-id
authority_id = "64b**"

# AAD application
kcsb = KustoConnectionStringBuilder.with_aad_application_key_authentication(cluster, client_id, client_secret, authority_id)

# # User/Password
# # In case you want to authenticate with AAD username and password
# username = "..."
# password = "..."
# kcsb = KustoConnectionStringBuilder.with_aad_user_password_authentication(cluster, username, password, authority_id)


# In case you want to authenticate with Azure CLI.
# Users are required to be in a logged in state in az-cli, for this authentication method to succeed. Run `az login` to login to azure cli.
# kcsb = KustoConnectionStringBuilder.with_az_cli_authentication(cluster)


# The authentication method will be taken from the chosen KustoConnectionStringBuilder.
#client = KustoClient(kcsb)


######################################################
##                       QUERY                      ##
######################################################

# once authenticated, usage is as following
db = "Orca"
query = "Orca3285Pack | take 10"

with KustoClient(kcsb) as client:
    try:
        response = client.execute(db, query)
    except Exception as err:
        mylog.exception(err)

# iterating over rows is possible
for row in response.primary_results[0]:
    # printing specific columns by index
    print("value at 0 {}".format(row[0]))
    print("\n")
```

## azure-kusto-data == 4.0.2
logs (az-login):
```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): adxvipanalytics.westeurope.kusto.windows.net:443
DEBUG:urllib3.connectionpool:https://adxvipanalytics.westeurope.kusto.windows.net:443 "GET /v1/rest/auth/metadata HTTP/1.1" 200 None
INFO:azure.identity._internal.decorators:AzureCliCredential.get_token succeeded
DEBUG:azure.identity._internal.decorators:[Authenticated account] Client ID: 04b***. Tenant ID: 64b***. User Principal Name: unavailableUpn. Object ID (user): 97***
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): adxvipanalytics.westeurope.kusto.windows.net:443
DEBUG:urllib3.connectionpool:https://adxvipanalytics.westeurope.kusto.windows.net:443 "POST /v2/rest/query HTTP/1.1" 200 None
```
trace:
```
```

## azure-kusto-data == 4.1.0
logs:
```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): adxvipanalytics.westeurope.kusto.windows.net:443
DEBUG:urllib3.connectionpool:https://adxvipanalytics.westeurope.kusto.windows.net:443 "GET /v1/rest/auth/metadata HTTP/1.1" 200 None
INFO:azure.identity._internal.decorators:AzureCliCredential.get_token succeeded
DEBUG:azure.identity._internal.decorators:[Authenticated account] Client ID: 04b***. Tenant ID: 64b***. User Principal Name: unavailableUpn. Object ID (user): 97***
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): adxvipanalytics.westeurope.kusto.windows.net:443
DEBUG:urllib3.connectionpool:https://adxvipanalytics.westeurope.kusto.windows.net:443 "POST /v2/rest/query HTTP/1.1" 400 0
```
trace:
```
ERROR:main:Server error response contains no data.
Traceback (most recent call last):
  File "c:\scratch\debug_adx\.venv\lib\site-packages\requests\models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
  File "C:\bin\Python\Python38\lib\json\__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "C:\bin\Python\Python38\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\bin\Python\Python38\lib\json\decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\scratch\debug_adx\.venv\lib\site-packages\azure\kusto\data\client.py", line 307, in _execute
    response_json = response.json()
  File "c:\scratch\debug_adx\.venv\lib\site-packages\requests\models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\scratch\debug_adx\dummy2.py", line 57, in <module>
    response = client.execute(db, query)
  File "c:\scratch\debug_adx\.venv\lib\site-packages\azure\kusto\data\client.py", line 160, in execute
    return self.execute_query(database, query, properties)
  File "c:\scratch\debug_adx\.venv\lib\site-packages\azure\core\tracing\decorator.py", line 78, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "c:\scratch\debug_adx\.venv\lib\site-packages\azure\kusto\data\client.py", line 175, in execute_query
    return self._execute(self._query_endpoint, database, query, None, self._query_default_timeout, properties)
  File "c:\scratch\debug_adx\.venv\lib\site-packages\azure\kusto\data\client.py", line 310, in _execute
    raise self._handle_http_error(e, endpoint, payload, response, response.status_code, response_json, response.text)
  File "c:\scratch\debug_adx\.venv\lib\site-packages\azure\kusto\data\client_base.py", line 118, in _handle_http_error
    raise KustoServiceError("Server error response contains no data.", response) from exception
azure.kusto.data.exceptions.KustoServiceError: Server error response contains no data.
```

**Comment by AsafMah** [comment]
I can't recreate it.

Are all of the variables really identical?

Did you create a virtualenv, installed 4.0.2, it worked, and then installed 4.1.0 with the same code and it didn;t?

**Comment by Gjermund06** [comment]
I have done some more debugging and tracing on this issue. For the records some of my colleague’s has also tried to reproduce the issue, without success.

In https://github.com/Azure/azure-kusto-python/blob/92d3e172823f7dcd68ce8f698ecf7a39b83c92ff/azure-kusto-data/azure/kusto/data/client.py#L281
the call to requests.session.post is wrapped in a KustoTracingAttributes

By inspecting the request_headers that is provided to POST, it include an 'x-ms-user' property.

For me this is is `'x-ms-user': 'GjermundBøthun'` 
By removing this property or removing the norwegian spacial charachter 'ø' it works.

The same ges for calling from postman or curl.

IE, this is acutal a ADX issue, which doesnt allow special charachters in headers.

But I guess it is a good pracasis to remove such charchters or encode them fromthis package.

**Comment by AsafMah** [comment]
Thank you, I will pass it forward to the service team. 
Closing for now.

**Comment by AsafMah** [comment]
Re-opening - we will handle this at an sdk level, will be in the next version

**Comment by yogilad** [comment]
This is being fixed in #469

**Comment by AsafMah** [comment]
Fixed in 4.1.4
